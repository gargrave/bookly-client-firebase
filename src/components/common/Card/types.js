// @flow
export type CardTextListItem = {
  title?: string,
  value?: string,
}
