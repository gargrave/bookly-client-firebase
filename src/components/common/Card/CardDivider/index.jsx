// @flow
import React from 'react';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
};

const CardDivider = ({}: Props) => {
  return (
    <hr className={buildClasses(['card__divider'])} />
  );
};

CardDivider.propTypes = {
};

export default CardDivider;
