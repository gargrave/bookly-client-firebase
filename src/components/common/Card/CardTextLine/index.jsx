// @flow
import React from 'react';
import { bool, string } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
  bold?: boolean,
  condensed?: boolean,
  text: string,
};

const CardTextLine = ({
  bold = false,
  condensed = false,
  text,
}: Props) => {
  return (
    <p
      className={buildClasses([
        'card__text',
        bold ? 'card__text--bold' : '',
        condensed ? 'condensed' : '',
      ])}
    >
      {text}
    </p>
  );
};

CardTextLine.propTypes = {
  bold: bool,
  condensed: bool,
  text: string.isRequired,
};

export default CardTextLine;
