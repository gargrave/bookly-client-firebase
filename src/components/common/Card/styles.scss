@import '../../../styles/_index';

.bookly-card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 600px;
  padding: 10px 20px;

  &--top-margin-med {
    margin-top: 50px;
  }

  hr {
    margin: 12px 0;
  }

  &--hoverable:hover {
    background: #e4effc;
    cursor: pointer;
  }

  &__header,
  &__title,
  &__text {
    margin: 0;
  }

  &__header,
  &__title {
    font-weight: bold;
  }

  &__header {
    font-size: 1.5rem;
  }

  &__text {
    color: #6e6969;
  }

  &__text + &__text {
    margin-top: 8px;
  }
}

.bookly-detail-card {
  padding: 15px;

  .bookly-card__title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .bookly-card__title + .bookly-card__text {
    font-size: 1.1rem;
  }
}

.bookly {
  &-list-detail-card {
    padding: 10px
  }
}
