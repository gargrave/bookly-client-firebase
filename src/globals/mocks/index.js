import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
