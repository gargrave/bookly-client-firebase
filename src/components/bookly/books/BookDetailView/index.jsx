// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Book } from '../../../../constants/flowtypes';

import { buildClasses } from '../../../../utils/cssHelpers';

import Alert from '../../../common/Alert';
import AuthorLink from '../../authors/AuthorLink';
import Button from '../../../common/Button';
import ButtonRow from '../../../common/ButtonRow';
import Card from '../../../common/Card';

type Props = {
  book: Book,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
  topLevelError?: string,
};

function BookDetailView({
  book,
  onBackClick,
  onDeleteClick,
  onEditClick,
  topLevelError,
}: Props) {
  const {
    created,
    title,
    updated,
  } = book;

  return (
    <div className={buildClasses('book-detail-view')}>
      {topLevelError &&
        <Alert
          message={topLevelError}
          type="danger"
        />
      }

      <Card
        classes={['card--top-margin-med', 'detail-card', 'book-detail-card']}
        hoverable={false}
        title={title}
      >
        <AuthorLink
          author={book.author}
        />

        <hr/>
        <p className={buildClasses('card-text')}>
          <strong>Added:</strong> {format(created, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
        <p className={buildClasses('card-text')}>
          <strong>Updated:</strong> {format(updated, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>

        <hr/>
        <ButtonRow>
          <Button
            onClick={onEditClick}
            position="left"
            text="Edit"
            type="info"
          />
          <Button
            onClick={onDeleteClick}
            text="Delete"
            type="danger"
          />
          <Button
            onClick={onBackClick}
            position="right"
            text="Back"
            type="light"
          />
        </ButtonRow>
      </Card>
    </div>
  );
}

BookDetailView.propTypes = {
  book: shape({
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    author: shape({
      firstName: string,
      lastName: string,
    }),
    title: string,
  }).isRequired,
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
  topLevelError: string,
};

export default BookDetailView;
