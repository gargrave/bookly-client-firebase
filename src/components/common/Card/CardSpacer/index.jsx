// @flow
import React from 'react';
import { oneOf } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

import './styles.css';

type Props = {
  size?: string,
};

const CardSpacer = ({
  size = 'medium',
}: Props) => {
  return (
    <div className={buildClasses(`card__spacer--${size}`)} />
  );
};

CardSpacer.propTypes = {
  size: oneOf(['small', 'medium', 'large']),
};

export default CardSpacer;
