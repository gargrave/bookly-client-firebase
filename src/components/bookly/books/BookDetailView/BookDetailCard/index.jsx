// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Book } from '../../../../../constants/flowtypes';

import AuthorLink from '../../../authors/AuthorLink';
import Button from '../../../../common/Button';
import ButtonRow from '../../../../common/ButtonRow';
import Card from '../../../../common/Card';
import CardSpacer from '../../../../common/Card/CardSpacer';
import CardTextList from '../../../../common/Card/CardTextList';

type Props = {
  book: Book,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
};

const bookDatesTextList = (book: Book) => {
  const {
    created,
    updated,
  } = book;

  return [
    { title: 'Added', value: format(created, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Updated', value: format(updated, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};

const BookDetailCard = ({
  book,
  onBackClick,
  onDeleteClick,
  onEditClick,
}: Props) => {
  const {
    title,
  } = book;

  return (
    <Card
      classes={['detail-card', 'book-detail-card']}
      header={title}
      hoverable={false}
    >
      <AuthorLink
        author={book.author}
      />

      <CardTextList textList={bookDatesTextList(book)} />
      <CardSpacer size='large' />

      <ButtonRow>
        <Button
          onClick={onEditClick}
          position="left"
          text="Edit"
          type="info"
        />
        <Button
          onClick={onDeleteClick}
          text="Delete"
          type="danger"
        />
        <Button
          onClick={onBackClick}
          position="right"
          text="Back"
          type="light"
        />
      </ButtonRow>
    </Card>
  );
};

BookDetailCard.propTypes = {
  book: shape({
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    author: shape({
      firstName: string,
      lastName: string,
    }),
    title: string,
  }).isRequired,
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
};

export default BookDetailCard;
