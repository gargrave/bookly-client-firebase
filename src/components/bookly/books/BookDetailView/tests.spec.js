import React from 'react';
import { shallow } from 'enzyme';

import { bookMocks } from '../../../../utils/mocks/';

import BookDetailView from './';
import Card from '../../../common/Card';

const testBook = bookMocks[0];

const defaultProps = Object.freeze({
  book: testBook,
  onBackClick: jest.fn(),
  onDeleteClick: jest.fn(),
  onEditClick: jest.fn(),
});

function getComponent(extraProps = {}) {
  const props = Object.assign({}, defaultProps, extraProps);
  return shallow(<BookDetailView {...props} />);
}

describe('BookDetailView', () => {
  let component;

  test('matches the snapshot', () => {
    component = getComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = getComponent();
    const parentClass = '.bookly-book-detail-view';
    expect(component.find(parentClass)).toHaveLength(1);
  });

  test('renders one Card component', () => {
    component = getComponent();
    expect(component.find(Card)).toHaveLength(1);
  });
});
