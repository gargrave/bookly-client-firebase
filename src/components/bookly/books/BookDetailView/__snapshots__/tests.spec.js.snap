// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetailView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookDetailView
    book={
      Object {
        "author": Object {
          "bookCount": 1,
          "created": 2017-07-01T01:16:43.881Z,
          "firstName": "Aldous",
          "id": "oiw223h3qndlksadffl3ksdf",
          "lastName": "Huxley",
          "updated": 2017-07-01T01:16:43.881Z,
        },
        "created": 2018-01-14T01:36:26.220Z,
        "id": "sv3hmbefpc95MRXQsTHr",
        "title": "Brave New World",
        "updated": 2018-01-14T01:36:26.220Z,
      }
    }
    onBackClick={[Function]}
    onDeleteClick={[Function]}
    onEditClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <BookDetailCard
          book={
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            }
          }
          onBackClick={[Function]}
          onDeleteClick={[Function]}
          onEditClick={[Function]}
        />,
      ],
      "className": "bookly-detail-view bookly-book-detail-view",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-14T01:36:26.220Z,
            "id": "sv3hmbefpc95MRXQsTHr",
            "title": "Brave New World",
            "updated": 2018-01-14T01:36:26.220Z,
          },
          "onBackClick": [Function],
          "onDeleteClick": [Function],
          "onEditClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <BookDetailCard
            book={
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              }
            }
            onBackClick={[Function]}
            onDeleteClick={[Function]}
            onEditClick={[Function]}
          />,
        ],
        "className": "bookly-detail-view bookly-book-detail-view",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            },
            "onBackClick": [Function],
            "onDeleteClick": [Function],
            "onEditClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
